src/Logic/Calculate.js

```diff
@@ -0,0 +1,92 @@
+import operate from './Operate';
+
+export default function calculate(data, buttonName) {
+  let {
+    total, next, operation, answer,
+  } = data;
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  const operators = ['+', 'x', '-', '÷'];
+
+  if (buttonName === '=') {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      answer = total;
+      next = null;
+      operation = null;
+      total = null;
+    } else if (answer && next && operation) {
+      total = operate(answer, next, operation);
+      answer = total;
+      next = null;
+      operation = null;
+      total = null;
+    } else if (total && answer && operation) {
+      total = operate(total, answer, operation);
+      answer = total;
+      next = null;
+      operation = null;
+      total = null;
+    }
+  } else if (buttonName === 'AC') {
+    total = null;
+    next = null;
+    operation = null;
+    answer = null;
+  } else if (buttonName === 'A') {
+    total = answer;
+    next = null;
+  } else if (buttonName === 'A' && operation) {
+    next = answer;
+  } else if (buttonName === '+/-') {
+    total *= -1;
+    next *= -1;
+    answer *= -1;
+  } else if (buttonName === '%') {
+    if (total && next && operation) {
+      next /= 100;
+    } else {
+      total /= 100;
```

Comment:
* [ ] [REQUIRED]  `operate` switches `operation` and uses `plus`, `minus`, `times`, `div` and `modifier` (turning the number into a percentage based number, `e.g. 95 % == .95`) functions ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png)).
As stated in the previous review and the guideline above, the modifier (%) should be handled in the `Operate.js` module. 
*Hint:* In this particular case, your second number is always *0.01* (i.e. `total` or `next` will be `null` when calling `operate`). Thus in `Operate.js`, you can have:
```
case '%':
      return one.times(0.01);
```